components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -1343,4 +1343,32 @@ private int readHighestPriorityOfPolicies() throws PolicyManagerDAOException {
         return priority;
     }
 
+    @Override
+    public int getPolicyCount() throws PolicyManagerDAOException {
+
+        Connection conn;
+        PreparedStatement stmt = null;
+        ResultSet resultSet = null;
+        int policyCount = 0;
+        try {
+            conn = this.getConnection();
+            String query = "SELECT COUNT(DM_POLICY.ID) FROM DM_POLICY";
+            stmt = conn.prepareStatement(query);
+            resultSet = stmt.executeQuery();
+
+            while (resultSet.next()) {
+                policyCount = resultSet.getInt(0);
+            }
+            return policyCount;
+
+        } catch (SQLException e) {
+            String msg = "Error occurred while reading the policies from the database.";
+            log.error(msg, e);
+            throw new PolicyManagerDAOException(msg, e);
+        } finally {
+            PolicyManagementDAOUtil.cleanupResources(stmt, resultSet);
+            this.closeConnection();
```

Comment:
Currently, PolicyManagementDAOFactory.closeConnection() is used to close the "Connection".
This feature is implemented in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/cdbdc86f42b350c24e9d059ff7c23e1f09242df4